File: wisefy/src/androidTest/java/com/isupatches/wisefy/test/IsWifiEnabledTests.java
package com.isupatches.wisefy.test;


import com.isupatches.wisefy.test.base.BaseTestClass;
import com.isupatches.wisefy.test.base.TestActivity;
import org.junit.Test;
import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.when;


public class IsWifiEnabledTests extends BaseTestClass<TestActivity> {

    public IsWifiEnabledTests() {
        super(TestActivity.class);
    }

    @Test
    public void failure() {
        setManagers();
        when(mMockWiFiManager.isWifiEnabled()).thenReturn(false);
        assertEquals(false, mWiseFy.isWifiEnabled());
    }

    @Test
    public void failure_nullWifiManager() {
        setManagersToNull();
        assertEquals(false, mWiseFy.isWifiEnabled());
    }

    @Test
    public void success() {
        setManagers();
        when(mMockWiFiManager.isWifiEnabled()).thenReturn(true);
        assertEquals(true, mWiseFy.isWifiEnabled());
    }
}

File: wisefy/src/test/java/com/isupatches/wisefy/IsWifiEnabledTests.kt
package com.isupatches.wisefy

import android.os.Build

import com.isupatches.wisefy.internal.base.BaseUnitTest

import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith

import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

/**
 * Tests the ability to determine if wifi is enabled on a device.
 *
 * @author Patches
 */
@RunWith(RobolectricTestRunner::class)
@Config(sdk = [Build.VERSION_CODES.M])
internal class IsWifiEnabledTests : BaseUnitTest() {

    @Test fun failure_prechecks() {
        mockWiseFyPrechecksUtil.isWifiEnabled_failure()
        assertEquals(false, wisefy.isWifiEnabled())
    }

    @Test fun failure() {
        mockWiseFyPrechecksUtil.isWifiEnabled_success()
        mockNetworkUtil.isWifiEnabled(false)
        assertEquals(false, wisefy.isWifiEnabled())
    }

    @Test fun success() {
        mockWiseFyPrechecksUtil.isWifiEnabled_success()
        mockNetworkUtil.isWifiEnabled(true)
        assertEquals(true, wisefy.isWifiEnabled())
    }
}
